src/test/java/com/puppycrawl/tools/checkstyle/filters/SuppressWithPlainTextCommentFilterTest.java

```diff
@@ -0,0 +1,485 @@
+            "messages.properties", "key", null, SeverityLevel.ERROR, null, getClass(), null);
+        final String fileName = "nonexisting_file";
+        final AuditEvent auditEvent = new AuditEvent(this, fileName, message);
+
+        final SuppressWithPlainTextCommentFilter filter = new SuppressWithPlainTextCommentFilter();
+
+        try {
+            filter.accept(auditEvent);
+            fail(IllegalStateException.class.getSimpleName() + " is expected");
+        }
+        catch (IllegalStateException ex) {
+            assertEquals("Invalid exception message",
+                "Cannot read source file: " + fileName, ex.getMessage());
+
+            final Throwable cause = ex.getCause();
+            assertTrue("Exception cause has invalid type",
+                cause instanceof FileNotFoundException);
+            assertEquals("Invalid exception message",
+                fileName + " (No such file or directory)", cause.getMessage());
+        }
+    }
+
+    @Test
+    public void testFilterWithCustomMessageFormat() throws Exception {
+        final DefaultConfiguration filterCfg =
+            createModuleConfig(SuppressWithPlainTextCommentFilter.class);
+        final String messageFormat =
+            ".*" + getCheckMessage(FileTabCharacterCheck.class, MSG_CONTAINS_TAB) + ".*";
+        filterCfg.addAttribute("messageFormat", messageFormat);
+
+        final DefaultConfiguration fileTabCheckCfg =
+            createModuleConfig(FileTabCharacterCheck.class);
+        fileTabCheckCfg.addAttribute("eachLine", "true");
+
+        final DefaultConfiguration regexpCheckCfg = createModuleConfig(RegexpSinglelineCheck.class);
+        regexpCheckCfg.addAttribute("id", "ignore");
+        regexpCheckCfg.addAttribute("format", ".*[a-zA-Z][0-9].*");
+
+        final String[] suppressed = {
+            "8:1: " + getCheckMessage(FileTabCharacterCheck.class, MSG_CONTAINS_TAB),
+        };
+
+        final String[] violationMessages = {
+            "6: " + getCheckMessage(RegexpSinglelineCheck.class, MSG_REGEXP_EXCEEDED,
+                ".*[a-zA-Z][0-9].*"),
+            "8:1: " + getCheckMessage(FileTabCharacterCheck.class, MSG_CONTAINS_TAB),
+            "8: " + getCheckMessage(RegexpSinglelineCheck.class, MSG_REGEXP_EXCEEDED,
+                ".*[a-zA-Z][0-9].*"),
+            "10: " + getCheckMessage(RegexpSinglelineCheck.class, MSG_REGEXP_EXCEEDED,
+                ".*[a-zA-Z][0-9].*"),
+            "13: " + getCheckMessage(RegexpSinglelineCheck.class, MSG_REGEXP_EXCEEDED,
+                ".*[a-zA-Z][0-9].*"),
+        };
+
+        verify(
+            createCustomChecker(filterCfg, fileTabCheckCfg, regexpCheckCfg),
+            getPath("InputSuppressWithPlainTextCommentFilterCustomMessageFormat.java"),
+            removeSuppressed(violationMessages, suppressed)
+        );
+    }
+
+    private Checker createCustomChecker(Configuration... childConfigs) throws Exception {
```

Comment:
@MEZk Similar here. Can we work around this?

Reply:
@rnveach 
So, your idea is to copy paste this logic in each test method. Why should I do this? I need to extract the logic which adds multiple children to Checker to separate method. I'm against copy-pasted logic in tests. Test should be as clean and easy to read as possible. Ideally they should consist of 3 parts: arrange, act, verify. If you have more convenient reasons to have the code copy-pasted, please, prove tour idea.